README.md

```diff
@@ -62,73 +57,112 @@ The final relationships consist of direct relationships and indirect relationshi
-
-The table schema developed in this stage should satisfy 2 main queries:
-
-1. **Q1:** Give me all relationships for obj_id
-2. **Q2:** Give me all objects that subscribe to relationship_value
-3. (**Q3:** Give me all objects that subscribe to relationship_type)
-
-![GitHub Logo](/images/schema.png)
-
-*Figure 3: Table View*
-
-The picture above represents the two main tables that should satisfy the queries from Stage 2.
-The original base table easily satisfies Q1. The table created through the MV can satisfy Q2.
-Even Q3 can be easily addressed with a slightly different MV.
 
-The relationship values for each primary relationship can be either direct values or references unique identifiers that can be used to query lookup tables for additional details on the relationship value. Lookup tables are generated for a specific subset of relationship values.
+*Table 1: Definitions for Final relationships*
 
 
 ## Implementation
 
-#### Offline Search and Training
-
-###### Query Component
-
-
-This component will look for atomic indicators of relationships. Atomic indicators are either identical values that can be shared by artefacts’ analytic results or calculated values that are used to provide some measure of similarity between artefacts. The Query component will generate the following relationship_types and values. These primary relationships will also have an assigned weight that can be used by the second stage of the process to calculate the final relationships.
-
- (relationship_type, service, relationship_value)  | weight_definition
- ------------------------------------------------- | -----------------
- (imphash_similar_to, PEInfo, imphash)  |   
- (pehash_similar_to, PEInfo, pehash) |  
- (signed_by, PEInfo, signature) |  
- (communicated_with_ip, CUCKOO, ip) |  
- (communicated_with_dom, CUCKOO, domain) |  
- (related_to_ip, DNSMeta, ip) |  
- (resolves_to_ip, DNSMeta, A Record )  |  
- (resolves_to_ip, DNSMeta, AAAA Record)  |  
- (related_to, DNSMeta, metadata)  |  
- (av_similar_to, VirusTotal, signature_similarity) |  
- (yara_similar_to, YARA, complex_AV_match) |  
-
-*Table 2: Definitions for primary relationships*
-
-###### ML Component
-
-This component will utilize ML algorithms to train models based on a labeled dataset and then assign every new unknown incoming artefact (depending on the type of artefact) to one of the trained malicious clusters/classes.
+### Offline Stage
+
+#### Knowledge Base Generation
+
+The data we use is provided by the Holmes-Totem and Holmes-Totem-Dynamic services.
+The service results contain a variety of data, some of which is useless to the  
```

Comment:
extra newline after services. I also wouldn't be so negative. You can just state that you identify and extract the pertinent information for identifying relationships.